packages/rest-hooks/src/resource/Resource.ts

```diff
@@ -30,13 +30,15 @@ export default abstract class Resource extends SimpleResource {
   ) {
     let options: RequestInit = {
       method: method.toUpperCase(),
-      headers: {
-        'Content-Type': 'application/json',
-        // "Content-Type": "application/x-www-form-urlencoded",  -- maybe use this if typeof body is FormData ?
-      },
     };
     if (this.fetchOptionsPlugin) options = this.fetchOptionsPlugin(options);
     if (body) options.body = JSON.stringify(body);
```

Comment:
what happens if i provide both `body` and `options.body`?

Reply:
body will override that. this is the legacy way of doing things. @rest-hooks/rest maps much cleaner to normal 'fetch' interface patterns, and has many more extension points to specify user-defined behavior.

i agree it's not obvious how this will work, thus the change in design.

as for why it was done this way - body is called on a case-by-case basis, whereas fetchOptionsPlugin() is defined for endpoint statically. So body takes precedent due to being defined in a case with more dynamic information.

e.g., every call to `fetch(params, body)` can specify an exact body, given a fetch, which is defined at same time as `fetchOptionsPlugin()`